src/hooks/index.tsx

```diff
@@ -70,6 +78,16 @@ export const PangolinWeb3Provider: FC<Web3ProviderProps> = ({
   );
 };
 
+export const MixPanelProvider: FC<MixPanelProviderProps> = ({ children, mixpanelToken }: MixPanelProviderProps) => {
```

Comment:
1. lets move mixpanel related code to separate `mixpanel.ts` file
2. create `useMixPanel` hook instead of doing `const isActiveMixPanel = useContext(MixPanelContext);` everywhere
3. in context lets create `track` function which is wrapper of mixpanel.track so we don't need to write `source: 'pangolin-components',` everytime
4. lets define actions const in mixpanel.ts for `'Wallet Connected'`, ``Swap`` etc actions

Reply:
@prodesert22 4th point is not handled yet.